lib/ansible/collections/list.py

```diff
@@ -26,6 +28,8 @@ def list_valid_collection_paths(search_paths=None, warn=False):
 
     if search_paths is None:
         search_paths = []
+    elif isinstance(search_paths, string_types):
+        raise TypeError("'search_paths' is string and expected list")
```

Comment:
If we're going to validate the type here (to catch programming errors, I assume), why not just use ``if not isinstance(search_paths, list)`` to catch all misuses?

Reply:
i should test for 'sequence' to include tuples/sets/etc but avoid strings